src/dvc_objects/fs/base.py

```diff
@@ -68,13 +68,17 @@ class FileSystem:
 
     PARAM_CHECKSUM: ClassVar[Optional[str]] = None
 
+    VERSION_QUERY_KEY: ClassVar[Optional[str]] = None
+
     def __init__(self, fs=None, **kwargs):
         self._check_requires(**kwargs)
 
         self.jobs = kwargs.get("jobs") or self._JOBS
         self.hash_jobs = kwargs.get("checksum_jobs") or self.HASH_JOBS
         self._config = kwargs
         self.fs_args = {"skip_instance_cache": True}
+        if "version_aware" in kwargs:
+            self.fs_args["version_aware"] = kwargs.pop("version_aware")
```

Comment:
Could we do this in particular filesystems that use it? E.g. dvc-azure, dvc-s3, etc

Reply:
If we think versioning is going to end up as implementation specific in fsspec then it would make sense to do it in the individual dvc fs plugins

But given that we are probably going to end up being responsible for doing all of the underlying fsspec versioning implementations (other than s3fs) I think it might make more sense for us to try and standardize this (based on s3fs)? And then on the DVC side we should only need to account for versioning in our base fsspec wrapper.